src/Microsoft.DocAsCode.Dfm.VscPreview/DfmProcess.cs

```diff
@@ -63,11 +66,36 @@ private static void ReceiveContent()
             Console.Write('\a');
         }
 
+        private static void JsonMarkupReceiveContent()
+        {
+            // a simple protocol(get String According to the numOfRow and connect them)
+            string numStr = Console.ReadLine();
+            int numOfRow = Convert.ToInt32(numStr);
+            StringBuilder markdownContent = new StringBuilder();
+            for (int i = 0; i < numOfRow; i++)
+            {
+                markdownContent.AppendLine(Console.ReadLine());
+            }
+
+            var result = JsonMarkup(markdownContent.ToString());
+
+            // append with customized endCode
+            Console.Write(result);
+            Console.Write('\a');
+        }
+
         private static string DfmMarkup(string basedir, string filename, string markdownContent)
         {
             DfmServiceProvider dfmServiceProvider = new DfmServiceProvider();
-            IMarkdownService dfmService = dfmServiceProvider.CreateMarkdownService(new MarkdownServiceParameters { BasePath = basedir });
+            IMarkdownService dfmService = dfmServiceProvider.CreateMarkdownService(new MarkdownServiceParameters {BasePath = basedir});
```

Comment:
now there maybe two or more editor use one childprocess to parse the markdown file,so everytime we need the basedir and path,but it makes sense in the Jsonmarkup

---
In reply to: [77452764](https://github.com/dotnet/docfx/pull/667#discussion_r77452764) [](ancestors = 77452764)